### ISSUE TITLE
unnecessary_mut_passed false positives in github.com/google/tarpc

### ISSUE THREAD

**Issue by tikue** [Manishearth/rust-clippy#1468]
[tarpc's](https://github.com/google/tarpc) `service!` macro generates traits like the following:

```rust
service! {
    rpc read(size: u32) -> Vec<u8>;
}

// Generates the following trait
trait SyncService {
    fn read(&self, size: u32) -> Vec<u8>;
}
```

Running clippy on [this code](https://github.com/google/tarpc/blob/master/examples/concurrency.rs) produces the following warning:

```
warning: The function/method "SyncService::read" doesn't need a mutable reference, #[warn(unnecessary_mut_passed)] on by default
  --> examples/concurrency.rs:32:1
   |
32 |   service! {
   |  _^ starting here...
33 | |     rpc read(size: u32) -> Vec<u8>;
34 | | }
   | |_^ ...ending here
```

I see two issues here:
1. There is no `&mut self` in `SyncService::read`
2. Should Clippy even be warning about `&mut self` in traits?

**Comment by Manishearth** [comment]
Right, the point is that someone is calling `read(&mut foo)` when `read(&foo)` would suffice

**Comment by tikue** [comment]
Ah, I see! That makes a lot more sense. Unfortunately the line numbers weren't helpful in figuring that out...
